monai/data/dataset.py

```diff
@@ -848,11 +851,12 @@ def _compute_cache(indices=None):
 
     def disable_share_memory_cache(self):
         """
-        If the cache content is multiprocessing share memory list, convert it to a regular ptython list.
-        Because multiprocessing ProxyList is not supported for the GPU caching, may need to explicitly diasble it.
+        If the cache content is a multiprocessing shared memory ListProxy, convert it to a regular python list.
+        Because multiprocessing ListProxy is not supported for the GPU caching,  explicitly disable it.
 
         """
-        self._cache = list(self._cache)
+        if self.runtime_cache and not self._is_dist:
```

Comment:
I have 2 suggestions here:
1. Maybe we don't need check "if self.runtime_cache" here? Is there any case?
2. Could you please help add a warning message when we can't disable shared memory cache?

Thanks in advance.

Reply:
okay, i've updated it, and showing a warning message